components/brave_wallet/browser/ethereum_provider_impl.cc

```diff
@@ -1233,13 +1276,13 @@ void EthereumProviderImpl::CommonRequestOrSendAsync(base::ValueView input_value,
   } else if (method == kWeb3ClientVersion) {
     Web3ClientVersion(std::move(callback), std::move(id));
   } else if (method == kEthSubscribe) {
-    std::string event_type;
-    if (!ParseEthSubscribeParams(normalized_json_request, &event_type)) {
+    base::Value params;
+    if (!ParseEthSubscribeParams(normalized_json_request, &params)) {
```

Comment:
`ParseEthSubscribeParams` should check if `params` is a list of one or two elements and return them. The first one should be a string, the second one should be a dict(if exists)

EthSubscribe gets them as args:
`EthereumProviderImpl::EthSubscribe(const std::string& event_type, absl::optional<base::Value::Dict> filter,...)` and I don't think we need to do any validation of filter's content(for example we still don't do full validation of address per spec *address (optional): either a string representing an address or an array of such strings.*) - let that be backend responsibility:

```
  } else if (event_type == kEthSubscribeLogs && filter) {
    const auto gen_res = generateHexBytes(eth_log_subscriptions_);

    if (std::get<0>(gen_res)) {
      eth_logs_tracker_.Start(base::Seconds(kLogTrackerDefaultTimeInSeconds));
    }

    eth_logs_tracker_.AddSubscriber(std::get<1>(gen_res),
                                    std::move(*filter));

    std::move(callback).Run(std::move(id), base::Value(std::get<1>(gen_res)),
                            false, "", false);
  } else {
```

Reply:
I agree with just letting RPC endpoint to validate/handle the filter keys.
Just a side note, it seems the RPC endpoint can support eth_subscribe logs without filter, so in additional to support filter, we probably want to keep what we currently support (without filter) too.

Reply:
fixed